components/automate-cli/cmd/chef-automate/config.go

```diff
@@ -119,47 +137,162 @@ func runShowCmd(cmd *cobra.Command, args []string) error {
 		then automate cluster ctl deploy will patch the config to automate
 	*/
+	/*
+		// if isA2HARBFileExist() {
+		// 	if !configCmdFlags.acceptMLSA {
+		// 		response, err := writer.Prompt(`If you have created any new bundles using upgrade commands and not deployed it,
+		// 		this command will deploy that new airgap bundle with patching of configuration.
+		// 		Press y to agree, n to to disagree? [y/n]`)
+		// 		if err != nil {
+		// 			return err
+		// 		}
+
+		// 		if !strings.Contains(response, "y") {
+		// 			return errors.New("canceled Patching")
+		// 		}
+		// 	}
+		// 	input, err := ioutil.ReadFile(args[0])
+		// 	if err != nil {
+		// 		return nil
+		// 	}
+		// 	err = ioutil.WriteFile(AUTOMATE_HA_AUTOMATE_CONFIG_FILE, input, 0644)
+		// 	if err != nil {
+		// 		writer.Printf("error in patching automate config to automate HA")
+		// 		return err
+		// 	}
+		// 	return executeDeployment(args)
+		// }
+	*/
+
 	if isA2HARBFileExist() {
-		if !configCmdFlags.acceptMLSA {
-			response, err := writer.Prompt(`If you have created any new bundles using upgrade commands and not deployed it, 
-			this command will deploy that new airgap bundle with patching of configuration. 
-			Press y to agree, n to to disagree? [y/n]`)
-			if err != nil {
-				return err
-			}
 
-			if !strings.Contains(response, "y") {
-				return errors.New("canceled Patching")
+		if configCmdFlags.frontend {
+			frontendIps := append(infra.Outputs.ChefServerPrivateIps.Value, infra.Outputs.AutomatePrivateIps.Value...)
+			writer.Printf("IPs: " + strings.Join(frontendIps, "") + "Path :" + args[0])
+			for i := 0; i < len(frontendIps); i++ {
+				executePatchOnRemote(sshUser, sshPort, sskKeyFile, frontendIps[i], args[0], "fe")
 			}
 		}
-		input, err := ioutil.ReadFile(args[0])
-		if err != nil {
-			return nil
+		if configCmdFlags.postgresql {
+			for i := 0; i < 1; i++ {
+				executePatchOnRemote(sshUser, sshPort, sskKeyFile, infra.Outputs.PostgresqlPrivateIps.Value[i], args[0], "pg")
+			}
+		}
+		if configCmdFlags.opensearch {
+			for i := 0; i < 1; i++ {
+				executePatchOnRemote(sshUser, sshPort, sskKeyFile, infra.Outputs.OpensearchPrivateIps.Value[i], args[0], "os")
+			}
 		}
-		err = ioutil.WriteFile(AUTOMATE_HA_AUTOMATE_CONFIG_FILE, input, 0644)
+	} else {
+
+		cfg, err := dc.LoadUserOverrideConfigFile(args[0])
 		if err != nil {
-			writer.Printf("error in patching automate config to automate HA")
+			return status.Annotate(err, status.ConfigError)
+		}
+		if err = client.PatchAutomateConfig(configCmdFlags.timeout, cfg, writer); err != nil {
 			return err
 		}
-		return executeDeployment(args)
-	}
-
-	if err = client.PatchAutomateConfig(configCmdFlags.timeout, cfg, writer); err != nil {
-		return err
 	}
 
 	writer.Success("Configuration patched")
 	return nil
 }
 
+func executePatchOnRemote(sshUser string, sshPort string, sshKeyFile string, ip string, path string, remoteType string) {
```

Comment:
wirte a generic function here ...pass command also as an args ..so it can be call from any module

Reply:
Updated the function to generic. NEED TO TEST